src/v/cluster/partition_manager.cc

```diff
@@ -162,10 +163,26 @@ ss::future<> partition_manager::stop_partitions() {
     co_await _gate.close();
     // prevent partitions from being accessed
     auto partitions = std::exchange(_ntp_table, {});
-    _raft_table.clear();
+
+    {
+        auto current = _raft_table.begin();
+        while (current != _raft_table.end()) {
+            current = _raft_table.erase(current, ++current);
+            co_await ss::coroutine::maybe_yield();
+        }
+    }
+
     // shutdown all partitions
-    co_await ss::parallel_for_each(
-      partitions, [this](auto& e) { return do_shutdown(e.second); });
+    co_await ss::max_concurrent_for_each(
+      partitions, 1024, [this](auto& e) { return do_shutdown(e.second); });
+
+    {
+        auto current = partitions.begin();
+        while (current != partitions.end()) {
+            current = partitions.erase(current, ++current);
+            co_await ss::coroutine::maybe_yield();
+        }
```

Comment:
thinking about current and ++current and erase return value seems overly complicated?

```c++
while (!partitions.empty()) {
  partitions.erase(partitions.begin());
  co_await ss::coroutine::maybe_yield();
}
```